src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/PreviewLinkPopover.js

```
@@ -0,0 +1,160 @@
+// @flow
+import React from 'react';
+import {action, computed, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import copyToClipboard from 'copy-to-clipboard';
+import Button from 'sulu-admin-bundle/components/Button';
+import {ResourceRequester} from 'sulu-admin-bundle/services';
+import {transformDateForUrl} from 'sulu-admin-bundle/utils/Date';
+import {translate} from 'sulu-admin-bundle/utils';
+import PreviewStore from './stores/PreviewStore';
+import previewLinkStyles from './preview-link.scss';
+import type {PreviewLink} from './types';
+
+type Props = {|
+    previewStore: PreviewStore,
+|};
+
+@observer
+class PreviewLinkPopover extends React.Component<Props> {
+    @observable previewLink: ?PreviewLink;
+    @observable loading: boolean;
+    @observable generating: boolean = false;
+    @observable copying: boolean = false;
+
+    componentDidMount() {
+        this.loadPreviewLink();
+    }
+
+    @action loadPreviewLink() {
+        const {
+            previewStore,
+        } = this.props;
+
+        this.loading = true;
+        ResourceRequester.get('preview_links', {
+            resourceKey: previewStore.resourceKey,
+            resourceId: previewStore.id,
+            locale: previewStore.locale,
+        }).then(action((previewLink) => {
+            this.previewLink = previewLink;
+            this.loading = false;
+        })).catch(action(() => {
+            this.loading = false;
```

this will swallow all errors - is that done intentionally?

---

yeah was copied from somewhere else i think 😂 will remove the catch block